new_study_assistant.py

```
@@ -0,0 +1,624 @@
+### load in packages
+import argparse
+import seaborn
+import random
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import Levenshtein
+import scipy
+import urllib, json
+from collections import Counter
+import math
+from scipy.spatial.distance import pdist, squareform
+from scipy.cluster.hierarchy import linkage, dendrogram, fcluster
+
+#define some functions
+#-----------------------------------
+#downloads main study information from the portal
+def get_all_study_info():
+    url = "http://www.cbioportal.org/api/studies"
+    response = urllib.urlopen(url)
+    json_data = json.loads(response.read())
+    return json_data
+
+#downloads all attribute data for a list of studies
+def get_attribute_data(study_list):
+    studies=[]
+    study_attributes = []
+
+    #loop over studies and download data from api
+    for study in study_list:
+        try:
+            studyID = study['studyId']
+        except:
+            studyID = study
+        study_api_url = 'http://cbioportal.org/api/studies/' + studyID + '/clinical-attributes'
+        df = pd.read_json(study_api_url)
+
+        #make sure the study is not empty
+        if not df.empty:
+            studies.append((studyID ,df))
+            study_attributes.append((studyID ,df['clinicalAttributeId'].tolist()))
```

maybe throw an exception here if the study is empty - meaning there's a bug either in this code or maybe the cbioportal api changed and this code needs to be updated